runtime/platform_isolate_manager.h

```diff
@@ -0,0 +1,50 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_RUNTIME_PLATFORM_ISOLATE_MANAGER_H_
+#define FLUTTER_RUNTIME_PLATFORM_ISOLATE_MANAGER_H_
+
+#include <atomic>
+#include <mutex>
+#include <unordered_set>
+
+#include "third_party/dart/runtime/include/dart_api.h"
+
+namespace flutter {
+
+/// Maintains a list of registered platform isolates, so that they can be
+/// proactively shutdown as a group during shell shutdown.
+class PlatformIsolateManager {
+ public:
+  /// Returns whether the PlatformIsolateManager is shutdown. New isolates
+  /// cannot be registered after the manager is shutdown. Callable from any
+  /// thread. The result may be obsolete immediately after the call, except when
+  /// called on the platform thread.
+  bool IsShutdown();
```

Comment:
You may consider having 2 methods, so it's obvious on the call site if the caller gets a racy return value.

```c++
// Only callable on the platform thread.
bool HasShutdown() {
  ASSERT(current-thread-running-on-platform-task-runner);
  ...
}

// If returns true guaranteed to be shutdown, if false it may or may not have shutdown.
bool HasShutdownWithFalseNegativePossibility() {
   ...
}
```

Reply:
ping :)

Reply:
Done.